### PR TITLE
[DOC] add documentation & doc tests to fsl preprocess

### PR DESCRIPTION
**Pull request by neuroumbrage**
Adding working doc tests and updated the documentation links:

- BET
- FAST
- McFLIRT
- SUSAN (doc links only)

### CODE REVIEW

File: .cache/v/cache/lastfailed [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (oesteban) [inline .cache/v/cache/lastfailed]: This file seems to be mistakenly added

File: nipype/algorithms/tests/test_auto_CompCor.py [hunk 2]
```diff
@@ -1,37 +0,0 @@
-# AUTO-GENERATED by tools/checkspecs.py - DO NOT EDIT
```
Comment (oesteban) [inline nipype/algorithms/tests/test_auto_CompCor.py]: This file should not be modified. Please rebase to current upstream and run ```make specs```.

File: nipype/algorithms/tests/test_auto_ComputeDVARS.py [hunk 3]
```diff
@@ -17,6 +17,8 @@ def test_ComputeDVARS_inputs():
     ),
     in_mask=dict(mandatory=True,
     ),
+    intensity_normalization=dict(usedefault=True,
```
Comment (oesteban) [inline nipype/algorithms/tests/test_auto_ComputeDVARS.py]: @chrisfilo is this something we forgot to update in master?

File: nipype/algorithms/tests/test_auto_ErrorMap.py [hunk 4]
```diff
@@ -1,35 +0,0 @@
-# AUTO-GENERATED by tools/checkspecs.py - DO NOT EDIT
```
Comment (oesteban) [inline nipype/algorithms/tests/test_auto_ErrorMap.py]: Again, no ```test_auto_ ``` should be modified in this PR. Please update to current master and we'll see if there is some deviation from master that caused these tests to be updated.

File: nipype/algorithms/tests/test_auto_FramewiseDisplacement.py [hunk 5]
```diff
@@ -19,6 +19,8 @@ def test_FramewiseDisplacement_inputs():
     ),
     out_file=dict(usedefault=True,
     ),
+    parameter_source=dict(mandatory=True,
```
Comment (oesteban) [inline nipype/algorithms/tests/test_auto_FramewiseDisplacement.py]: @chrisfilo, another auto test that needs update?

File: nipype/interfaces/freesurfer/tests/test_auto_BBRegister.py [hunk 6]
```diff
@@ -18,6 +18,7 @@ def test_BBRegister_inputs():
     usedefault=True,
     ),
     init=dict(argstr='--init-%s',
+    mandatory=True,
```
Comment (oesteban) [inline nipype/interfaces/freesurfer/tests/test_auto_BBRegister.py]: @effigies is this something missing in current master?

File: nipype/testing/data/ev_test_condition_0_1.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
+0.000000 10.000000 1.000000
```
Comment (oesteban) [inline nipype/testing/data/ev_test_condition_0_1.txt]: This file should not be added.

File: nipype/interfaces/freesurfer/tests/test_auto_BBRegister.py (hunk 6 above)
Comment (effigies) [inline nipype/interfaces/freesurfer/tests/test_auto_BBRegister.py]: This was added on a system with FreeSurfer v5.3. The interface changed with v6.0, and we've set the tests to follow v6.0 (see #1811 and #1817).

TLDR This change should be reverted.

**Comment by neuroumbrage** [comment]
I rebased and ran `make specs`. A lot of the test files are no longer changed, however, auto tests for BBRegister, ApplyXFM, and ApplyXfm are all changed. Should I commit my updated version with the changed auto tests or without?

**Comment by effigies** [comment]
Only update auto tests where they reflect changes you've made in the interfaces they're testing. I don't think you've changed any, so I'd leave all of them out.

**Comment by satra** [comment]
@neuroumbrage - would you mind closing this PR and creating a new PR. because of some github mechanics this pull-request shows a lot of changes that are already in master.

http://stackoverflow.com/questions/16306012/github-pull-request-showing-commits-that-are-already-in-target-branch
